extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -495,7 +498,10 @@ export class AmpA4A extends AMP.BaseElement {
         .then(returnedArray => returnedArray[0]);
       });
     }))
-    .then(returnedArray => returnedArray[0]);
+    .then(returnedArray => {
```

Comment:
Woo.  I did look at it before -- should have written back.  That promise machinery is a bit gnarly.  It's got two nested iterations via `.some()`.  At most points, if I inserted a ping, I'd get either one ping per provider or one ping per key.  This was the only point I could figure out to put something where it would only give back a single ping.

But I don't think this way of doing it is actually any overhead.  There is already a `.then` there to extract the 0 element of the array:
```javascript
.then(returnedArray => returnedArray[0]);
```

I'm just augmenting the anonymous function by one call:
```javascript
    .then(returnedArray => {
      this.emitLifecycleEvent('adResponseValidateEnd');
      return returnedArray[0];
    });
```